bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,291 @@
+A new maven plugin and goal (wildfly-jar:package) is developed to output a bootable jar during maven build.
+The bootable jar is composed of:
+
+* Zipped server provisioned thanks to galleon (slim or fat server).
+* User application (war, jar, ear) if not a hollow bootable jar.
+* Runtime to launch server.
+* The plugin output a jar named <project>-wildfly.jar
+
+===== Bootable jar runtime resolution
+
+The set of classes that implement the boot logic are retrieved at packaging time. 
+The new “org.wildfly.core:wildfly-jar-boot” jar artifact (implementation located in wildfly-core repository) 
+is resolved thanks to the wildfly-core version of the provisioned server and shaded into the bootable jar. 
+If the artifact doesn’t exist, the packaging will abort, the server being packaged not supporting bootable 
+jar packaging. (eg: old release not containing new artifacts).
+
+This artifact is in charge to unzip the server and setup a JBOSS_HOME. Once setup, 
+it setups JBoss module classloader and call into a new JBoss module “org.wildfly.bootable-jar” 
+that is in charge of the actual server run.  This new JBoss module implementation located in wildfly-core, 
+(“org.wildfly.core:wildfly-jar-runtime” artifact), is expected to be automatically provisioned by galleon in all cases. 
+If the module is not part of the provisioned server, the packaging aborts, 
+the server being packaged not supporting bootable jar packaging (eg: old release not containing new artifacts).
+
+===== Galleon provisioning
+
+Galleon provision a server thanks to the WildFly Galleon feature-pack.
+Galleon provisioning is operated inside the maven plugin. There is 2 configuration ways:
+
+* Galleon configuration as maven plugin configuration items:
+** A Galleon feature-pack location that identifies the server and version. This is required if no provisioning.xml file is provided nor feature-packs list.
+** A list of feature-packs (if no feature-pack location has been set).
+** Optionally, a list of layers to include.
+** Optionally, a list of layers to exclude.
+* Galleon provisioning.xml file located in <app src>/galleon/provisioning.xml. NB: maven plugin configuration items (if set) override provisioning.xml file. 
+Using provisioning.xml file, custom feature-packs can be combined during the provisioning phase to provision modules, features, content (eg: DB drivers).
+
+If no galleon layers are specified, then a configuration identical to the default _standalone-microprofile.xml_ is provisioned.
+
+===== CLI script execution
+
+When building the bootable jar, a path to a set of CLI scripts can be provided to update the server configuration packaged in the bootable jar. The CLI scripts are concatenated and executed once the server is provisioned and deployment copied into it (if any).
+Having multiple CLI scripts provides enough flexibility to reuse scripts in different contexts.
+The plugin can be configured with a CLI properties file to resolve properties present in scripts.
```

Comment:
Is  this documented somehow? How are those properties going to be used? Which should be the location and the name pof the properties file? Are there any other property resolution mechanism?

Reply:
I managed to find an article that covers it, I linked it. The CLI covers it in its help.